### PR TITLE
CIV-6870 refactor ccd config upload script

### PR DESCRIPTION
**Pull request by filippo-possenti-version1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-6870

### Change description ###
Refactored CCD config upload script as follows:

- added explicit support for aat, prod and staging environments
- added error for unknown environments
- removed unused (wrongly applied) "excludeNonProdFiles" argument
- removed duplicated handling of "-nonprod.json" files

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by sabahirfan — APPROVED** [review]
Please update PR title, otherwise LGTM.

**Review by kurt-uk — APPROVED** [review]

File: bin/build-release-ccd-definition.sh [hunk 1]
```diff
@@ -16,11 +15,16 @@ elif [ ${environment} == demo ]; then
 elif [ ${environment} == local ]; then
   # upload doesn't currently work with this command due to SDO and SDO-HNL files
   excludedFilenamePatterns="-e *-prod.json"
-elif [ ${excludeNonProdFiles} == true ]; then
-  excludedFilenamePatterns="-e UserProfile.json,*-nonprod.json,*-COS-nonprod.json,*GAspec.json,*-GAR2GAspec-nonprod.json,*DJ.json,*DJspec.json,*DJ-SDO-nonprod.json,*DJ-nonprod.json,*CUI.json,*CUI-nonprod.json,*-GAR3CP-nonprod.json"
+elif [ ${environment} == aat ]; then
+  excludedFilenamePatterns="-e UserProfile.json,*-nonprod.json,*GAspec.json,*DJ.json,*DJspec.json,*CUI.json"
```
Comment (sampankumar) [inline bin/build-release-ccd-definition.sh:19]: We are missing SDO related files here

**Review by PaulH-HMCTS — APPROVED** [review]

File: bin/build-release-ccd-definition.sh (hunk 1 above)
Comment (filippo-possenti-version1) [reply bin/build-release-ccd-definition.sh:19]: No, we're not.

If you notice on the left, the exclusion has `*-nonprod.json`, which would automatically include `*DJ-SDO-nonprod.json`.
The bit on the right it's a simplification, as all those matching `*-nonprod.json` were already excluded. If anything, this simplification should prevent mistakes.

**Review by sankhajuria — APPROVED** [review]

**Comment by filippo-possenti-version1** [comment]
All leads are now aware. Merging as there's enough approvals.

**Pull request merged**